nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -74,179 +42,51 @@ BtNavigator::~BtNavigator()
 }
 
 nav2_util::CallbackReturn
-BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
+BtNavigator::on_configure()
 {
-  RCLCPP_INFO(get_logger(), "Configuring");
-
-  // use suffix '_rclcpp_node' to keep parameter file consistency #1773
-  auto options = rclcpp::NodeOptions().arguments(
-    {"--ros-args",
-      "-r", std::string("__node:=") + get_name() + "_rclcpp_node",
-      "--"});
-  // Support for handling the topic-based goal pose from rviz
-  client_node_ = std::make_shared<rclcpp::Node>("_", options);
```

Comment:
Should client_node_ be in the generic class? I think its only used by self_client_ used in Bt navigator so it should be in this object.

Reply:
`client_node_` is also the `rclcpp::Node::SharedPtr` object that's put on the blackboard for the behavior tree plugins to use.
https://github.com/ros-planning/navigation2/blob/eff208ef5aa1b26356bda8f5fa8f304c7884327c/nav2_bt_navigator/src/bt_navigator.cpp#L124